docs/topics/native/native-migration-guide.md

```diff
@@ -0,0 +1,94 @@
+
+This guide compares the new [Kotlin/Native memory manager](native-memory-manager.md) with the legacy one and
+describes how to migrate your projects.
+
+The most noticeable change in the new memory manager is lifting restrictions on object sharing. You don't
+need to freeze objects to share them between threads, specifically:
+
+* Top-level properties can be accessed and modified by any thread without using `@SharedImmutable`.
+* Objects passing through interop can be accessed and modified by any thread without freezing them.
+* `Worker.executeAfter` no longer requires operations to be frozen.
+* `Worker.execute` no longer requires producers to return an isolated object subgraph.
+* Reference cycles containing `AtomicReference` and `FreezableAtomicReference` do not cause memory leaks.
+
+Apart from easy object sharing, the new memory manager also brings other major changes:
+
+* Global properties are initialized lazily when the file they are defined in is accessed first. Previously global
+  properties were initialized at the program startup. As a workaround, you can mark
+  properties that must be initialized at the program start with the `@EagerInitialization` annotation. Before using, check
+  its [documentation](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.native/-eager-initialization/).
+* `by lazy {}` properties support thread-safety modes and do not handle unbounded recursion.
+* Exceptions that escape `operation` in `Worker.executeAfter` are processed like in other runtime parts, by trying to
+  execute a user-defined unhandled exception hook or terminating the program if the hook was not found or failed with
+  an exception itself.
+* Freezing is deprecated, disabled by default, and will be removed in future releases.
+  Do not use freezing if you don't need your code to work with the [legacy memory manager](#support-both-new-and-legacy-memory-managers).
+
+Follow these guidelines to migrate your projects from the legacy memory manager:
+
+## Update Kotlin
+
+The new Kotlin/Native memory manager has been enabled by default since Kotlin 1.7.20. Check the Kotlin version and [update
+to the latest one](releases.md#update-to-a-new-release) if necessary.
+
+## Update dependencies
+
+<deflist style="medium">
+    <def title="kotlinx.coroutines">
+        <p>Update to version 1.6.0 or later. Do not use versions with the <code>native-mt</code> suffix.</p>
+        <p>There are also some specifics with the new memory manager you should keep in mind:</p>
+        <ul>
+            <li>Every common primitive (channels, flows, coroutines) works through the Worker boundaries, since freezing is not required.</li>
+            <li>Unlike the <code>native-mt</code> version, library objects are transparent for freeze. For example, if you freeze a channel, all of its internals get frozen, so it won't work as expected. In particular, this can happen when freezing something that captures a channel.</li>
```

Comment:
I think this point can be dropped because dropping is now disabled by default with the new MM.

Reply:
OK, removed this item